### PR TITLE
Improved test coverage for loader

### PR DESCRIPTION
**Pull request by hechth**
- Improved the test coverage of the loader class by loading metabolomics data and the strain mapping.
- preparing for GNPS loader implementation
- added hashing function to `Strain` class

### CODE REVIEW

File: tests/test_loader.py [hunk 1]
```diff
@@ -1,24 +1,89 @@
...
+            "root": DATA_DIR / "extracted",
+            "platform_id": "",
+            "overrides": {
+                "strain_mappings_file": str(DATA_DIR / "strain_mappings.csv")
+            }
+        }
+    }
+    shutil.rmtree(
+        DATA_DIR /
+        "extracted")
```
Comment (CunliangGeng) [inline tests/test_loader.py]:
```suggestion
    shutil.rmtree(DATA_DIR / "extracted")
```

**Review by CunliangGeng — APPROVED** [review]
Looks good to me.

**Review by gcroci2 — APPROVED** [review]
I couldn't follow all the changes since great part of the code base is still a black box for me, but the parts I understood look good! Be sure to fix the static typing and the tests before merging.

**Comment by hechth** [comment]
@gcroci2 one test is currently broken due to the gcf_id - not much to do here :/

**Review by CunliangGeng — APPROVED** [review]
Thanks, it's ok to merge.

**Review by gcroci2 — APPROVED** [review]

**Comment by CunliangGeng** [comment]
It's fixed by temporarily using gcf.id. See commit: https://github.com/NPLinker/nplinker/commit/b0cfcbd3ea9d908393e7d8ecebcff863d42db280
> @gcroci2 one test is currently broken due to the gcf_id - not much to do here :/

**Pull request merged**